File: bookOfProblems/codeforces.com/edu/Двоичный поиск/A. Двоичный поиск.js
#!/usr/bin/env node

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let commands = '';

process.stdin.on('data', function (input) {
    commands += input;
});

process.stdin.on('end', function () {

    /*
     * Code Start
     **************************************************************************/

    let input = commands.trim().split('\n');
    let [n, k] = input[0].trim().split(' ').map(Number);
    let arr1 = input[1].trim().split(' ').map(Number);
    let arr2 = input[2].trim().split(' ').map(Number);

    for (let i = 0; i <= arr2.length; i++) {
        if (binarySearch(arr1, arr2[i]) === false) console.log("NO")
        else console.log("YES")
    }
});

function binarySearch(array, target) {

    let left = 0;
    let right = array.length - 1

    while (left <= right) {
        let middle = Math.floor(left + ((right - left) / 2));
        let guess = array[middle]

        if (guess === target) {
            return middle
        } else if (guess > target) {
            right = middle - 1
        } else if (guess < target) {
            left = middle + 1
        }
    }
    return false
}

File: bookOfProblems/codeforces.com/edu/Двоичный поиск/A. Двоичный поиск.py
arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

n, k = map(int, input().split())
arr = list(map(int, input().split()))
k_arr = list(map(int, input().split()))


def binary_search(arr, x):
    # берём отрезок, с нулевого до последнего элемента
    l = 0
    r = len(arr) - 1
    
    # на этом отрезке ищем элемент Х. Пока отрезок не пустой
    while l <= r: # иди r >= l, не важно
        # мы тыкаемся в середину. Объявляем миддл внутри цикла.
        m = l + (r - l) // 2
        if arr[m] == x:
            return "YES"
        # если элемент меньше Х, это значит что все элементы левее ещё меньше, поэтому надо искать с m + 1
        if arr[m] < x:
            l = m + 1
        if arr[m] > x:
            r = m - 1
    # Если while закончился, то значит мы отрезок проверили и он остался пустым, выводим None
    return "NO"


for n in k_arr:
    print(binary_search(arr, n))
